pkg/networkutils/network.go

```diff
@@ -126,9 +126,34 @@ func isDuplicateRuleAdd(err error) bool {
 	return strings.Contains(err.Error(), "File exists")
 }
 
+// find out the primary interface name
+func findPrimaryInterfaceName(primaryMAC string) (string, error) {
+
+	log.Debugf("Trying to find primary interface that has mac : %s", primaryMAC)
+
+	interfaces, err := net.Interfaces()
```

Comment:
This call to list all interfaces is probably the biggest change. 😄

Reply:
@mogren I just unit tested the `net.Interfaces()` call on ubuntu t3.large instance. It is working as expected.
Here is the test program
```
cat test.go
package main

import (
	"fmt"
	"net"
)

func main() {

	interfaces, _ := net.Interfaces()

	for _, intf := range interfaces {
		fmt.Printf("interface %v, mac %v\n", intf.Name, intf.HardwareAddr)
	}
}
```
Here is the test output
```
root@ip-172-31-0-186:/home/ubuntu# ./test
interface lo, mac 
interface ens5, mac 0a:5f:92:e3:59:86
```
Here is the output of ifconfig
```
root@ip-172-31-0-186:/home/ubuntu# ifconfig
ens5: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9001
        inet 172.31.0.186  netmask 255.255.240.0  broadcast 172.31.15.255
        inet6 fe80::85f:92ff:fee3:5986  prefixlen 64  scopeid 0x20<link>
        ether 0a:5f:92:e3:59:86  txqueuelen 1000  (Ethernet)
        RX packets 6072  bytes 7297970 (7.2 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 1699  bytes 199849 (199.8 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 191  bytes 14956 (14.9 KB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 191  bytes 14956 (14.9 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```